hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ScannerContext.java

```diff
@@ -110,7 +110,13 @@
    */
   final ServerSideScanMetrics metrics;
 
-  ScannerContext(boolean keepProgress, LimitFields limitsToCopy, boolean trackMetrics) {
+  /**
+   * Set this to true will prevent the looping skip of heap cells, see HBASE-25709.
```

Comment:
I read this PR first w/o looking at the JIRA. I did not understand why I would want to avoid 'looping skip of heap cells' after reading comments like this. I'd suggest say more here in the code on what this flag does? Thanks.

Reply:
Thank, @saintstack . I'll explain more here.